### PR TITLE
pnpm, pnpx: add page

### PR DESCRIPTION
**Pull request by Unbuttun**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/common/pnpm.md
Comment (marchersimon) [inline pages/common/pnpm.md:5]:
```suggestion
> More information: <https://pnpm.io>.
```

File: pages/common/pnpm.md
Comment (marchersimon) [inline pages/common/pnpm.md:11]:
```suggestion
- Download all the packages listed as dependencies in `package.json`:
```

File: pages/common/pnpm.md
Comment (marchersimon) [inline pages/common/pnpm.md:23]: Do you have to execute both of the commands? If so, we should write
 `pnpm install {{module_name}} --save-dev && pnpm install -D {{module_name}}`

Also, we use long arguments where possible:
```suggestion
`pnpm install {{module_name}} --save-dev`

`pnpm install --dev {{module_name}}`
```

File: pages/common/pnpm.md
Comment (marchersimon) [inline pages/common/pnpm.md:27]: Are you sure, this example is correct? I can't find it in <https://pnpm.io/cli/install>.

File: pages/common/pnpm.md
Comment (marchersimon) [inline pages/common/pnpm.md:39]: Same here, is `-g` maybe some kind of globally available flag?

File: pages/common/pnpx.md
Comment (marchersimon) [inline pages/common/pnpx.md:11]:
```suggestion
- Execute a specific binary from a given npm module, in case the module has multiple binaries:
```

File: pages/common/pnpx.md
Comment (marchersimon) [inline pages/common/pnpx.md:13]: Is there a long option for `-p`?

File: pages/common/pnpx.md
Comment (marchersimon) [inline pages/common/pnpx.md:15]:
```suggestion
- Show help:
```
This is what we use in most other pages.

**Review by marchersimon — CHANGES_REQUESTED** [review]

File: pages/common/pnpm.md
Comment (Unbuttun) [reply pages/common/pnpm.md:23]: Nah it was just 2 types on how you can install dev dependencies. Either by doing `--dev` or `--save-dev` or by simply `-D`

File: pages/common/pnpm.md
Comment (Unbuttun) [reply pages/common/pnpm.md:27]: yep it is, I've used it multiple times, also whenever pnpm updates, the command they suggest us is also this `pnpm install -g pnpm`

File: pages/common/pnpm.md
Comment (Unbuttun) [reply pages/common/pnpm.md:39]: yeah it is same as npm's `-g` flag

File: pages/common/pnpx.md
Comment (Unbuttun) [reply pages/common/pnpx.md:13]: long option?

File: pages/common/pnpm.md
Comment (marchersimon) [reply pages/common/pnpm.md]: I think that most clients don't support having two command lines. We have to remove one of the commands.

File: pages/common/pnpx.md
Comment (marchersimon) [reply pages/common/pnpx.md:13]: I mean `--package`.

Comment (Unbuttun) [reply pages/common/pnpx.md:13]: Yeah it do have `--package` for `-p`

File: pages/common/pnpm.md
Comment (Unbuttun) [reply pages/common/pnpm.md]: Yeah in here, I removed the the `--save-dev` one and only kept `-D` one, should i change it to `--dev`? but i think `-D`is more than enough

File: pages/common/pnpm.md
Comment (marchersimon) [reply pages/common/pnpm.md:37]: Then we should replace it with `--global`

File: pages/common/pnpm.md
Comment (marchersimon) [inline pages/common/pnpm.md:21]:
```suggestion
`pnpm install --dev {{module_name}}`
```

File: pages/common/pnpm.md
Comment (marchersimon) [reply pages/common/pnpm.md]: We generally prefer the long variants where possible, so users understand the command instead of just remembering it.

File: pages/common/pnpm.md
Comment (Unbuttun) [reply pages/common/pnpm.md:37]: nah i don think we should replace it, because almost nowhere anyone uses --global but everyone uses -g be it tutorials or anything, so there is no need to be extra verbose

File: pages/common/pnpm.md
Comment (bl-ue) [inline pages/common/pnpm.md:3]: I don't like when package managers say "for JavaScript", because it's really just for Node.js
```suggestion
> Fast, disk space efficient package manager for Node.js.
```

File: pages/common/pnpx.md
Comment (bl-ue) [inline pages/common/pnpx.md:4]:
```suggestion
> Directly execute binaries from npm packages, using `pnpm` instead of `npm`.
```

File: pages/common/pnpm.md
Comment (bl-ue) [inline pages/common/pnpm.md:31]:
```suggestion
- Print a tree of locally installed modules:
```

File: pages/common/pnpm.md
Comment (marchersimon) [reply pages/common/pnpm.md:37]: @bl-ue what do you think?

Comment (sbrl) [reply pages/common/pnpm.md:37]: Generally in tldr pages we prefer to use the long-form arguments, as this aids clarity - so I'd suggest using the long-form here. However, on the `npm` page we do this:

```markdown
- Download a package and install it globally:

`npm install -g {{module_name}}`
```

....so this may be ok - but I'd suggest updating the description to read `- List top-level [g]lobally installed modules:` though, just in case.

**Review by marchersimon — APPROVED** [review]
Thanks @Unbuttun

File: pages/common/pnpm.md
Comment (Unbuttun) [reply pages/common/pnpm.md:37]: Can you suggest the change? I'm currently on mobile so can't edit stuff easily

File: pages/common/pnpm.md
Comment (marchersimon) [inline pages/common/pnpm.md:35]:
```suggestion
- List top-level [g]lobally installed modules:
```

File: pages/common/pnpm.md
Comment (marchersimon) [reply pages/common/pnpm.md:37]: [Done](https://github.com/tldr-pages/tldr/pull/5863/files#r624548363)

File: pages/common/pnpx.md
Comment (sbrl) [inline pages/common/pnpx.md:12]: This is confusing. The language in the example talks about binaries, but then it mentions a package name in the example? :thinking:

Comment (navarroaxel) [reply pages/common/pnpx.md:12]: @sbrl We have the same example in the `npx` page https://github.com/tldr-pages/tldr/blob/main/pages/common/npx.md

File: pages/common/pnpx.md
Comment (navarroaxel) [inline pages/common/pnpx.md:14]:
```suggestion
- Display help:
```

File: pages/common/pnpm.md
Comment (bl-ue) [reply pages/common/pnpm.md:25]: Could we use `--global` here?  Or is it preferrable not to?

**Review by navarroaxel — APPROVED** [review]

**Review by bl-ue — APPROVED** [review]

**Comment by Unbuttun** [comment]
So should I tag some maintainer to merge it?

**Comment by navarroaxel** [comment]
> So should I tag some maintainer to merge it?

No @Unbuttun, just we try to keep a 2 days window before merge. Maybe another maintainer have something to say. But I think it looks good, thanks! 😁

File: pages/common/pnpx.md
Comment (sbrl) [reply pages/common/pnpx.md:12]: Ah, ok. I must not be as familiar with npx as I thought :woman_shrugging:

**Review by sbrl — APPROVED** [review]
Thanks, @Unbuttun - looks good to me.

Generally, we have the following principles when reviewing / merging:

 - Leave the pull request open for at least 2 days to give people a chance to review
 - Merge when all maintainers who reviewed have then approved
 - ....unless their comments are clearly addressed / they are inactive and a reasonable time window has passed
 - At least 2 approvals are required to merge unless it's a trivial change

**Pull request merged**
